src/controllers/update/update.manage.trusts.orchestrator.controller.ts

```diff
@@ -60,6 +61,38 @@ const shouldGoToReviewTheTrust = (trustInReview: Trust) => {
   return !trustInReview.review_status?.reviewed_trust_details;
 };
 
+export const trustChangeHandler = async (req: Request, res: Response, next: NextFunction) => {
+  logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+  try {
+    const trustId = req.params[ROUTE_PARAM_TRUST_ID];
+    const appData: ApplicationData = getApplicationData(req.session);
+
+    await putTrustInChangeScenario(req, appData, trustId);
+
+    res.redirect(UPDATE_MANAGE_TRUSTS_REVIEW_THE_TRUST_URL);
+  } catch (error) {
+    next(error);
+    return;
+  }
+};
+
+const putTrustInChangeScenario = async (req: Request, appData: ApplicationData, trustId: string) => {
+  const trustForChangeScenario = appData.trusts?.splice(appData.trusts.findIndex(trust => trust.trust_id === trustId), 1)[0];
+  appData.update?.review_trusts?.push(trustForChangeScenario as Trust);
```

Comment:
do we need these "as" casts?

Reply:
potentially not, I cant remember but I usually do it to get around those compilation errors that'll comeplain you can't pass object of type 'Trust | undefined' into the push(), although we know (via middleware) that a user can't get to this route without having a valid trust to edit.

Reply:
It's because the splice is on appData.trusts? that it knows undefined is possible.

Reply:
findIndex could return -1. So probably worth checking for. I saw a similar splice in moveTrustOutOfReview which also assumes something is found - that comes from a user input. All user input should be handled carefully.

Reply:
It's not that it hasn't been thought of, that's the point of the middleware, it checks that the user hasn't passed in a garbage trustId that doesn't exist because the navigation check that runs is `navigation.hasTrustWithIdUpdate`.
So if a user tries to access 'change-handler-url-or-whatever/1234' they'll be redirected if the ID doesn't exist.

Reply:
Excellent!